pyo3-macros-backend/src/method.rs

```diff
@@ -599,24 +632,32 @@ impl<'a> FnSpec<'a> {
                         }
                     })
                     .collect();
-                let call = rust_call(args, &mut holders);
+                let (call, self_arg_span) = rust_call(args, &self_e, &mut holders);
+                let extract_gil_ref =
+                    quote_spanned! { self_arg_span => #self_e.extract_gil_ref(); };
```

Comment:
It'll be nice to refactor this all again a bit later to reduce duplication. I haven't got an obvious idea how though, so I'm happy to move ahead like this in time for beta / final release 👍